fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm

```diff
@@ -0,0 +1,132 @@
+/obj/item/hunting_contract
+	name = "\improper Hunter's Contract"
+	desc = "Should I have my lawyer read this?"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "scroll2"
+	w_class = WEIGHT_CLASS_SMALL
+	///have we claimed our weapon?
+	var/bought = FALSE
+	///the datum containing all weapons
+	var/datum/hunter_market/shop
+	///the weapon that we have purchased
+	var/selected_item
+	///the owner of this contract
+	var/datum/antagonist/monsterhunter/owner
+
+/obj/item/hunting_contract/Initialize(mapload, datum/antagonist/monsterhunter/hunter)
+	. = ..()
+	shop = new /datum/hunter_market
+	if(hunter)
+		owner = hunter
+
+/obj/item/hunting_contract/ui_interact(mob/living/user, datum/tgui/ui)
+	if(!IS_MONSTERHUNTER(user))
+		to_chat(usr, span_notice("What the hell do these symbols mean?"))
```

Comment:
Same as above, be a bit more objective with it. "You are unable to decipher the symbols." as an example